src/transaction.ts

```diff
@@ -6,35 +6,40 @@ import {
   TransactionBuilder,
   xdr,
 } from "stellar-base";
+import { SorobanRpc } from "./soroban_rpc";
 
 // TODO: Transaction is immutable, so we need to re-build it here. :(
 export function assembleTransaction(
   raw: Transaction | FeeBumpTransaction,
   networkPassphrase: string,
-  simulated: Array<null | {
-    footprint: Buffer | string | xdr.LedgerFootprint;
-    auth: Array<Buffer | string | xdr.ContractAuth>;
-  }>,
+  simulation: SorobanRpc.SimulateTransactionResponse,
 ): Transaction {
   if ("innerTransaction" in raw) {
     // TODO: Handle feebump transactions
     return assembleTransaction(
       raw.innerTransaction,
       networkPassphrase,
-      simulated,
+      simulation,
     );
   }
 
-  if (simulated.length !== raw.operations.length) {
+  if (
+    raw.operations.length !== 1 ||
+    raw.operations[0].type !== "invokeHostFunction"
+  ) {
     throw new Error(
-      "number of simulated operations not equal to number of transaction operations",
+      "unsupported operation type for soroban, must be only one InvokeHostFunctionOp in the tx.",
     );
   }
 
   // TODO: Figure out a cleaner way to clone this transaction.
   const source = new Account(raw.source, `${parseInt(raw.sequence, 10) - 1}`);
-  const txn = new TransactionBuilder(source, {
-    fee: raw.fee,
+  const txnBuilder = new TransactionBuilder(source, {
+    fee: (
+      parseInt(raw.fee, 10) ||
```

Comment:
not sure on the algorithm for applying tx fees, we get a single `minResourceFee` from simulation which is a roll-up of all suggested resource fees, and it looked like that is to be treated additively to the existing classic fee, so just taking this simple approach to start, but need insights if this should change.

Reply:
The resource fees aren't "optional". It's guaranteed to fail if we don't meet these.
I'd suggest:
```js
max(0 + parseInt(simulation.minResourceFee, 10), 0 + parseInt(raw.fee, 10)).toString()
```

Reply:
@tsachiherman , I see the [inclusion fee requirement commented on preflight pr](https://github.com/stellar/soroban-tools/pull/636#discussion_r1190088092), how should we best integrate notion of inclusion fee in downstream UX, I see a couple variations on options:

1. soroban-client retains same tx helper interface, the rpc simulation response includes `suggestedInclusionFee`, which can be applied automatically within `assembleTransaction()` as final tx.fee = max(suggestedInclusionFee + minResourceFee, tx.fee), callers check resulting tx.fee to decide if they want to submit or handle otherwise. 

2. soroban-client retains same interface, sets final tx.fee to max(minResourceFee, tx.fee) and we document that callers must add a minimum additional inclusionFee of x stroops to the tx.fee before submit

3. soroban-client interfaces are changed to allow callers to pass `inclusionFee` on `assembleTransaction()` and `prepareTransaction()` and final tx.fee is set to max(inclusionFee + minResourceFee, tx.fee), callers check resulting tx.fee to decide whether to submit or handle otherwise.

Reply:
I think that #1 is the best option, since it would end up with the developer already having a good fees being placed within the transaction. The developer can always change that later on, but out-of-the-box, it would "just work".

Reply:
ok, I've added new `suggestedInclusionFee` to the [rpc simulation response model here](https://github.com/stellar/js-soroban-client/pull/77/files#diff-718319220445fb0cf5b6fe5c0d0c88085f8f32ee0a90d41b496037b00836b8c3R124) and applying sum with `minResourceFee` to tx.fee automatically if is greater.

[mentioned need for this new attribute on rpc server side pr for preflight changes](https://github.com/stellar/soroban-tools/pull/636/files#r1190606646)

Reply:
this has course corrected a bit, landed with just summing tx.fee and simulation `minResourceFee`

https://github.com/stellar/js-soroban-client/pull/77#discussion_r1194296683